### PR TITLE
"Clear All Filters" button added

### LINKED ISSUE #551: ✨ [Feature request]:  Add a separate "Clear All filters" Button
Hi @atapas,

**The solution:**
If you want to remove all the applied filters, just click on "Clear All" button to get back the original list of plays (unfiltered).
Therefore, such button should be added to the filter modal.

Tell me if you are okay with position of "Clear All" button.

**Additional context**

react-play-feature-SS

### PR DESCRIPTION
**Pull request by mayukh551**
Hi @atapas,

# Description

On Filter Modal, "Clear All" button is added to remove all applied filters on the filter modal.

File FilterPlays.jsx
- Line 278: clearFilters() function added to remove all applied filter queries
- Another prop called "onClearFilters" is passed to filter Modal

File index.jsx which has the filter Modal
- CSS added to design the "Clear All" button
- onClick prop added to the same div to call clearFilters()

Fixes #551 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

I am new to Open Source.
So apologies in advance for any foolish attempts.

### CODE REVIEW

File: src/common/search/FilterPlays.jsx [hunk 1]
```diff
@@ -11,294 +11,322 @@ import { FormControl, MenuItem, Select } from "@mui/material";
...
+        setFilterQuery(modifiedFilterQuery);
+        setnoOfAppliedFilter(getAppliedFilter(modifiedFilterQuery));
+        if (location.pathname !== "/plays") {
+            navigate("/plays", { replace: true });
+        }
+        showModal && setShowModal(false);
+    };
+
+    // function to clear all filters applied
+    const clearFilters = () => {
```
Comment (mayukh551) [inline src/common/search/FilterPlays.jsx:278]: the function I have added that removes all applied filters

File: src/common/search/FilterPlays.jsx [hunk 2]
```diff
@@ -11,294 +11,322 @@ import { FormControl, MenuItem, Select } from "@mui/material";
...
-      >
-        {noOfAppliedFilter === 0 ? null : (
-          <div className="badge">{noOfAppliedFilter}</div>
-        )}
+    return (
+        <div className="search-filter">
+            <Modal
+                title="Filter Plays By"
+                onClose={() => setShowModal(false)}
+                onClearFilters={clearFilters}
```
Comment (mayukh551) [inline src/common/search/FilterPlays.jsx:300]: clearFilters function prop passed for to filter Modal

**Comment by atapas** [comment]
@mayukh551 Please check why cypress tests are failing?

**Comment by atapas** [comment]
@mayukh551 I do no see the plays in the preview build

image

**Comment by mayukh551** [comment]
Okay, I am looking into it.

**Comment by atapas** [comment]
> Open.Source.ReactPlay.Bug-report.1.mp4 
> 
> I have checked and it works fine in localhost. But I do not know why it is failing to render the play in the preview build.
> 
> Cypress tells me that the test fails at "Header component should render properly" But when hosted locally using "npm start", all components work just fine.

Hey, wanna join the discord as other folks will be available to help?


The link is in the footer of the home page, you can ask help under the needs help channel.

**Comment by mayukh551** [comment]
Sure!
I have already joined as Shelby_1917.
Looking forward to learn more about open source from you guys.

**Comment by atapas** [comment]
@mayukh551 Why closed?

**Pull request closed without merging**